src/App.tsx

```diff
@@ -1,26 +1,259 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { sendErrorMessage } from './components/errorsUnderFooter';
+import { useRef } from 'react';
+import { Footer } from './components/footer';
+import { ErrorMessage } from './components/errorsUnderFooter';
+import { MainSection } from './components/section';
+import { Header } from './components/header';
+import { Filter } from './types/Filter';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<Filter>(Filter.All);
+  const [title, setTitle] = useState<string>('');
+  const [loader, setLoader] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [deletingTodos, setDeletingTodos] = useState<number[]>([]);
+  const [toogleAllTodos, setToogleAllTodos] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(data => {
+        setTodoList(data);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [tempTodo, deletingTodos]);
+
+  const filterTodos = useMemo(() => {
+    switch (selectedFilter) {
+      case Filter.Active:
+        return todoList.filter(todo => !todo.completed);
+      case Filter.Completed:
+        return todoList.filter(todo => todo.completed);
+      case Filter.All:
+      default:
+        return todoList;
+    }
+  }, [todoList, selectedFilter]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  // відповідає за зміну статусу у ОДНОГО туду
+  function handleToggleCompletion(todoId: number) {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it everywhere
```suggestion
  const handleToggleCompletion = (todoId: number) =>{
```